app/javascript/components/tooltips/Concept.tsx

```diff
@@ -0,0 +1,244 @@
+      break
+    default:
+      throw new Error('unhandled tooltip show action')
+  }
+
+  if (nextState.show) {
+    addOpenTooltip(id, dispatch)
+  } else {
+    removeOpenTooltip(id)
+  }
+
+  // request-focus and focus actions take precedence over mouse events
+  // so it can close any other for accessibility purposes
+  if (
+    (source === 'request-focus' || source === 'focus') &&
+    nextState.show &&
+    hasOpenTooltip(id)
+  ) {
+    closeOtherOpenTooltips(id)
+  }
+
+  return nextState
+}
+
+function dispatch(
+  id = '',
+  dispatcher: React.Dispatch<any>, // eslint-disable-line @typescript-eslint/no-explicit-any
+  source: ShowActionSource,
+  action?: ShowActionValue
+): void {
+  dispatcher([id, dispatcher, source, action])
+}
+
+export const Concept = ({
+  endpoint,
+  parent,
+  hoverRequestToShow,
+  focusRequestToShow,
+  parentSlug = '',
+}: {
+  endpoint: string
+  parent: HTMLElement | null
+  hoverRequestToShow: boolean
+  focusRequestToShow: boolean
+  parentSlug?: string
+}): JSX.Element => {
+  const tooltipId = `concept-tooltip${
+    parentSlug ? `-${parentSlug}` : parentSlug
+  }`
+  // Retrieve the HTML contents from the endpoint
+  const { isLoading, isError, data: html_content } = useQuery(tooltipId, () =>
+    fetch(endpoint).then((res) => res.text())
+  ) as { isLoading: boolean; isError: boolean; data: string }
+
+  const tooltipRef = useRef(null)
+  const [{ show }, showDispatcher] = useReducer(showReducer, {
+    requestHover: false,
+    requestFocus: false,
+    hover: false,
+    focus: false,
+    show: false,
+  })
+
+  const popper = usePopper(parent, tooltipRef.current, {
+    placement: 'bottom',
+    modifiers: [{ name: 'offset', options: { offset: [0, 8] } }],
+  })
+
+  // For the next two useEffects:
+  // If I ignore the exhaustive deps warning and only have requestToShow, it
+  // works.  If I have the exhaustive deps added (popper) it causes inf loop
+  // due to popper reference changing on every render.
+
+  useEffect(() => {
+    if (hoverRequestToShow && popper?.update) {
+      popper.update()
+      dispatch(tooltipId, showDispatcher, 'request-hover', 'show')
+      return
+    }
+
+    setTimeout(
+      () => dispatch(tooltipId, showDispatcher, 'request-hover', 'hide'),
+      150
+    )
+  }, [hoverRequestToShow]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (focusRequestToShow && popper?.update) {
+      popper.update()
+      dispatch(tooltipId, showDispatcher, 'request-focus', 'show')
+      return
+    }
+
+    dispatch(tooltipId, showDispatcher, 'request-focus', 'hide')
+  }, [focusRequestToShow]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  // Get styles and classes to apply
+  const styles = { ...popper.styles.popper, zIndex: 10 }
+  const classNames = ['c-tooltip', 'c-concept-tooltip']
+  if (!show) {
+    classNames.push('tw-invisible')
+  }
+  if (show) {
+    classNames.push('tw-pointer-events-auto')
+  }
```

Comment:
Two comments here.

1. If/else

```suggestion
  if (show) {
    classNames.push('tw-pointer-events-auto')
  }
  else {
    classNames.push('tw-invisible')
  }
```

2. Do we need to turn `tw-pointer-events-auto` off? If the element has `display:none`, then I don't think it matter if these are on. So I think this can just be added to the default set on L200, and you can *only* have the `if(!show)..` bit

3. Is there a reason you're using `tw-invisible` and not `tw-hidden`? I think we should use `display:none` here unless there's a good reason not to.

2. I think this should have a default state and a pivot. I'm presuming we consider tooltips to be shown by default, so we can just `tw-pointer-events-auto`, that should just

Reply:
Also, if we have (2) on always (as opposed to requiring conditional logic), that should be added to the CSS class for `c-concept-tooltip` not added here..

Reply:
1) Sure, I think they were separated by something else which I removed
2) So, `pointer-events: none` is set on the figure, so it is inherited by the tooltip, and strangely, without it set to none seems to trigger it to show even when hidden.
3) Using **invisible** so that popper can calculate the position of the tooltip even when not seen and to make it invisible to screen-readers (rather than opacity)
4) I hear what you're saying, but I think what I'm saying is that without manipulating the pointer events (as in `(2)`) it causes some triggering problems.

Reply:
> without it set to none seems to trigger it to show even when hidden.

My guess would be because it's `invisible` and not `hidden`

> so that popper can calculate the position of the tooltip even when not seen

Why is this desirable?

Reply:
This system has been changed a bit, I can't really comment on this implmentation now.  Will explain when the PR is ready again for re-review

Reply:
So now, if the tooltip is not hovered, the div is not mounted to the DOM, when it is, it is initially invisible (visibility: hidden) then made visible.